### PR TITLE
Bo react

### PR DESCRIPTION
**Pull request by bopeng95**
# Built with create-react-app
## How to start program
- First `npm install` the dependencies
- run `yarn start` to start react app
## Fake api 
- Fake api is created inside `db.json` file, change as you wish
- To change the server, go to `json-server.json` and change the port number

More info: [json-server](https://github.com/typicode/json-server)
### Proxy
- in `package.json` there is a key called `"proxy"`
- you can change that to any endpoint domain you want it to be so if you want to call axios, you dont need to type out the domain from server
### Login, PlayList, Tracks page
- Login page is where you will login to your account
- ATM it is just the link to the playlist route where your playlist is displayed
- each playlist will link you to the tracks with up and down buttons as well as the count

### CODE REVIEW

File: client/betterblend/src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (theseanathan) [inline client/betterblend/src/App.test.js]: Is this file necessary?

File: client/betterblend/src/Home.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import './css/playlist.css';
+import axios from 'axios';
+
+class Home extends Component {
+
+	constructor() {
+		super();
+		this.state = {
+			classnames: '',
```
Comment (theseanathan) [inline client/betterblend/src/Home.js]: What is this for? It's not getting set from what I've seen

File: client/betterblend/src/components/Login.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import '../index.css';
+
+class Login extends Component {
+
+	constructor() {
+		super();
+		this.state = {
```
Comment (theseanathan) [inline client/betterblend/src/components/Login.js]: Same as above

File: client/betterblend/src/Home.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+	getPlaylists = () => {
+		axios.get('/api')
+		.then(data => this.setState({playlists: data.data.playlists}));
+	};
+
+	render() {
+		return (
+			<div className={this.state.classnames}>
+				{this.state.playlists.map(p =>
```
Comment (theseanathan) [inline client/betterblend/src/Home.js]: I've been trying to think of a better way to handle this because often the playlist may not have an image.
I'll put out a PR in a bit to show you my solution but I'm not a huge fan.
It's essentially a filter before the return statement.

File: client/betterblend/src/components/Track.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	getPlaylists = () => {
+		axios.get('/api')
+		.then(data => this.setState({tracks: data.data.tracks}));
+	};
+
+	render() {
+		return (
+			<div className={this.state.classnames}>
```
Comment (theseanathan) [inline client/betterblend/src/components/Track.js]: Also wondered, why is it `<div className={this.state.classnames}>` and not `<div class={this.state.classnames}>`

Comment (bopeng95) [reply client/betterblend/src/components/Track.js]: in react it is className not class, and each component has constructor and inside theres 'state' in which u can modify and put it inside html like `{this.state.<whatever>}` and you can setState as well.

File: client/betterblend/src/Home.js (hunk 4 above)
Comment (bopeng95) [reply client/betterblend/src/Home.js]: no worries I will remove that and make it into something else, but pics for tracks ok?

Comment (theseanathan) [reply client/betterblend/src/Home.js]: do you have any ideas on how to handle that? i think we should merge first and do an iteration after

File: client/betterblend/src/components/Track.js (hunk 5 above)
Comment (theseanathan) [reply client/betterblend/src/components/Track.js]: yaya i get the state component, i mean in the div object. why is it `className` and not `class`?

Comment (bopeng95) [reply client/betterblend/src/components/Track.js]: it has to be `className` inside any tag in react, `class=` doesnt work
[stackoverflow link](https://stackoverflow.com/questions/46989454/class-vs-classname-in-react-16)
